pkg/k8s/policyhelper/kind.go

```diff
@@ -0,0 +1,50 @@
+package policyhelper
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	gwv1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+	gwv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
+)
+
+type GroupKind struct {
+	Group string
+	Kind  string
+}
+
+func ObjGroupKind(obj client.Object) GroupKind {
```

Comment:
nit: Could this be renamed to ObjToGroupKind (same for other funcs below), given that it's a mapper method?